Document mixed-mode authentication and authorization (#1126)
neo4j/docs-operations: modules/ROOT/pages/authentication-authorization/ldap-integration.adoc

First, you configure Neo4j to use LDAP as an authentication and authorization provider.

. Make sure security is turned on.
For this purpose uncomment the setting `dbms.security.auth_enabled=false` and change its value to `true`.
. Uncomment the settings `dbms.security.authentication_providers` and `dbms.security.authorization_providers` and change their value to `ldap`.
This way, the LDAP connector is used as a security provider for both authentication and authorization.
+
Note these configuration values are comma-separated lists, so if you wish to continue to use native authentication and authorization alongside LDAP, then these providers can be added to the existing `native` provider:
+
.Configuration
======
[source,configuration,role="noheader"]
----
dbms.security.authentication_providers=ldap,native
dbms.security.authorization_providers=ldap,native
----

> This example contains a list of LDAP and Neo4j native providers to use for authentication.

Reviewer: Maybe we should add that the providers are queried in-order

Author: done!

First, you configure Neo4j to use LDAP as an authentication and authorization provider.

> . Make sure security is turned on.

Reviewer 2: A best practice is to use only words that count, so I wouldn't change step 1. In the case of OpenID Connect, it is "Make sure security is turned on.", because they don't need to do anything, just to verify that it is "true" (which is the default).

For this purpose uncomment the setting `dbms.security.auth_enabled=false` and change its value to `true`.
. Uncomment the settings `dbms.security.authentication_providers` and `dbms.security.authorization_providers` and change their value to `ldap`.
This way, the LDAP connector is used as a security provider for both authentication and authorization.
+

> Note these configuration values are comma-separated lists, so if you wish to continue to use native authentication and authorization alongside LDAP, then these providers can be added to the existing `native` provider.

Reviewer 2: ```suggestion
If you want, you can still use the `native` provider for mixed-mode authentication and authorization.
The values are comma-separated and queried in the declared order.
```

> The providers will be queried in the order they are declared.

Reviewer 2: ```suggestion
```

Reviewer 2: Because the sentence above contains the information.

Author: The sentence above tells only that you can include different auth providers in the list but it doesn't tell in what oder those providers will be queried: first native, then LDAP or in vice versa. Or the order doesn't mean anything.

Reviewer 2: Ah, I see what you mean. So, if you accept my suggestion:
"If you want, you can still use the `native` provider for mixed-mode authentication and authorization.
The values are comma-separated and queried in the declared order."
It contains the information and  "The providers will be queried in the order they are declared." is not needed.

+

> .Configuration

Reviewer 2: ```suggestion
.Configure Neo4j to use LDAP and the native authentication and authorization provider.
```

======
[source,configuration,role="noheader"]
----
dbms.security.authentication_providers=ldap,native
dbms.security.authorization_providers=ldap,native
----

> This example contains a list of LDAP and Neo4j native providers to use for authentication.

Reviewer 2: I don't think we need an explanation after the example. We either do it before or better as a caption.
```suggestion
```
